codegen/templates.ts

```diff
@@ -257,11 +291,19 @@ function tfheSolidityOperator(bits: number, operator: Operator, os: OverloadSign
       using {${fname} as ${operator.binarySolidityOperator}} for euint${bits} global;
     `);
 
-    res.push(`
+    if (!isTeeOperation) {
+      res.push(`
       function ${fname}(euint${bits} lhs, euint${bits} rhs) pure returns (euint${bits}) {
         return TFHE.${operator.name}(lhs, rhs);
       }
     `);
+    } else {
+      res.push(`
+      function ${fname}(euint${bits} lhs, euint${bits} rhs) pure returns (euint${bits}) {
+        return TEE.${operator.name}(lhs, rhs);
+      }
+    `);
+    }
```

Comment:
I think there's too much repetition in this PR. My initial idea was to do something like:

```js
const libName = isTeeOperation ? 'TEE' : 'TFHE'

  res.push(`
      function ${fname}(euint${bits} lhs, euint${bits} rhs) pure returns (euint${bits}) {
        return ${libName}.${operator.name}(lhs, rhs);
      }
    `);
```

almost all functions in this file can be refactored using this trick, which will make the code more readable.